python/lsst/daf/butler/cli/cmd/commands.py

```diff
@@ -455,3 +456,17 @@ def query_dimension_records(**kwargs):
         table.pprint_all()
     else:
         print("No results. Try --help for more information.")
+
+
+@click.command(cls=ButlerCommand)
+@repo_argument(required=True)
+@query_datasets_options(showUri=False, useArguments=False, repo=False)
+@destination_argument(help="Destination URI of folder to receive file artifacts.")
+@transfer_option()
+@click.option("--preserve-path/--no-preserve-path", is_flag=True, default=True,
+              help="Preserve the datastore path to the artifact at the destination.")
+@options_file_option()
+def retrieve_artifacts(**kwargs):
+    """Retrieve file artifacts associated with datasets in a repository."""
+    transferred = script.retrieveArtifacts(**kwargs)
+    print(f"Number of artifacts retrieved into destination {kwargs['destination']}: {len(transferred)}")
```

Comment:
Is this intentionally a `print()` as opposed to a log or anything else?

Might it be useful for the list of artifact URIs to be an (optional) output?

Reply:
print is what @n8pease seemed to use in other click commands so I followed that. I don't think we use logging at all at the click layer. I can move it down to the "script" layer and use `log.info` for it so that it can be turned off completely.

I was a bit worried about printing all the final URIs out but I can add a verbose mode to do that. Debug logging will show the explicit "transferring A to B" message.

Reply:
No need to use log.  Seems reasonable to have this be a print.  (Actually I should have removed that first question once I saw that you actually checked the output in the tests.)

Reply:
Looks like click wants me to use `click.echo()` -- https://click.palletsprojects.com/en/7.x/utils/#printing-to-stdout

(which also lets me use colors in the output)

Reply:
I'll leave it as `print` for now since it is probably a bit odd for this subcommand to use `click.echo` but all the others to use `print`.